pkg/client/oci/oci.go

```diff
@@ -354,9 +425,85 @@ func (r *Repo) Reload() error {
 	return errors.Errorf("reload method is not supported yet")
 }
 
-func (r *Repo) newDockerResolver() remotes.Resolver {
+
+// OciReference contains info about an oci reference
+type ociReference struct {
+	path string
+	tag string
+}
+
+// splitOciReference split the tag or digest from an oci reference if exists
+func splitOciReference(ociRef string) ociReference {
+	parts := strings.Split(ociRef, "@")
+	if len(parts) == 2 { // Digest provided (Easy to split)
+		digest := parts[1]
+		digestRef := strings.Split(digest, ":")
+		return ociReference{path: parts[0], tag: digestRef[1]}
+	}
+	parts = strings.Split(ociRef, ":") // Tag provided or host:port. Let's use a regex
+	if len(parts) >= 2 {
+		res := tagRegex.FindStringSubmatch(ociRef)
+		return ociReference{path: res[1], tag: res[2]}
+	}
+
+	return ociReference{path: ociRef}
+}
+
+// ociReferenceExists checks if a given oci reference exists in the repository
+// OCI endpoints: https://github.com/opencontainers/distribution-spec/blob/main/spec.md#endpoints
+func ociReferenceExists(repoUrl *url.URL, ociRef, username, password string, insecure bool) (bool, error) {
+	ociReference := splitOciReference(ociRef)
```

Comment:
by using go-container-registry it seems you'll get access to some of these check methods all of them using image.reference as receiver

https://pkg.go.dev/github.com/google/go-containerregistry@v0.7.0/pkg/v1/remote#Head

Reply:
+1 if there is a lib that manages it